### PR TITLE
Auth deprecated methods removal

### PR DESCRIPTION
**Pull request by jonsimantov**
Remove methods that were deprecated in the 11.0.0 release last year.
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Review by a-maurice — CHANGES_REQUESTED** [review]
Going to need to update the Auth unit tests (the stuff under auth/tests) too

File: auth/src/include/firebase/auth.h [hunk 1]
```diff
@@ -50,7 +50,6 @@ struct AuthCompletionHandle;
 struct AuthResultCompletionHandle;
```
Comment (a-maurice) [inline auth/src/include/firebase/auth.h:50]: I think you delete AuthCompletionHandle in this change, so remove it here too

File: auth/src/desktop/auth_providers/federated_auth_provider.cc [hunk 2]
```diff
@@ -216,7 +137,7 @@ const char* CheckForRequiredAuthenicatedUserData(
 }
 
 // Helper function which uses the AuthCompletionHandle to plumb an
```
Comment (a-maurice) [inline auth/src/desktop/auth_providers/federated_auth_provider.cc:139]: AuthResultCompletionHandle

**Review by a-maurice — COMMENTED** [review]
Another sanity check to try, run the reference doc generation, and make sure that looks good.

**Review by a-maurice — APPROVED** [review]

**Comment by jonsimantov** [comment]
> Another sanity check to try, run the reference doc generation, and make sure that looks good.

Makes sense - running packaging workflow to generate docs from.

**Comment by jonsimantov** [comment]
> Another sanity check to try, run the reference doc generation, and make sure that looks good.

Confirmed that docs generation looks good, merging this.

**Pull request merged**